File: Best Time to Buy and Sell Stock II.py
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/submissions/

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        # Exit function if "prices" only has 1 element
        if len(prices) == 1:
            return 0
        
        # Iterate list
        totalProfit = 0
        for idx in range(1, len(prices)): # NOTE <= Starting at 2 because loop will check for previous element
            
            # If current price is greater than previous price, add profit to totalProfit
            if prices[idx] > prices[idx - 1]:
                totalProfit += prices[idx] - prices[idx - 1]
                
        return totalProfit

File: Best Time to Buy and Sell Stock II.java
/*
https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/
https://youtu.be/3SJ3pUkPQMc?si=j0LGPamJPD_cMKWm

Time Complexity: O(n) where n = length of array

Space Complexity: O(1)
*/

class Solution {

    public int maxProfit(int[] prices) {
        if (prices.length == 1) {
            return 0;
        }

        int profit = 0;
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] > prices[i - 1]) {
                profit += prices[i] - prices[i - 1];
            }
        }

        return profit;
    }
}
